File: raffia-core/src/main/java/com/codepoetics/raffia/operations/Setters.java
package com.codepoetics.raffia.operations;

import com.codepoetics.raffia.baskets.ArrayContents;
import com.codepoetics.raffia.baskets.Basket;
import com.codepoetics.raffia.baskets.ObjectEntry;
import com.codepoetics.raffia.baskets.PropertySet;

import java.math.BigDecimal;
import java.util.Collection;
import java.util.Map;

public final class Setters {

  private Setters() {
  }

  public static Updater toBasket(Basket value) {
    return Updaters.toConstant(value);
  }

  public static Updater toString(String value) {
    return toBasket(Basket.ofString(value));
  }

  public static Updater toNumber(BigDecimal value) {
    return toBasket(Basket.ofNumber(value));
  }

  public static Updater toBoolean(boolean value) {
    return toBasket(Basket.ofBoolean(value));
  }

  public static Updater toNull() {
    return toBasket(Basket.ofNull());
  }

  public static Updater toArray(Collection<Basket> contents) {
    return toBasket(Basket.ofArray(contents));
  }

  public static Updater toArray(ArrayContents contents) {
    return toBasket(Basket.ofArray(contents));
  }

  public static Updater toArray(Basket...contents) {
    return toBasket(Basket.ofArray(contents));
  }

  public static Updater toObject(ObjectEntry...properties) {
    return toBasket(Basket.ofObject(properties));
  }

  public static Updater toObject(Map<String, Basket> properties) {
    return toBasket(Basket.ofObject(properties));
  }

  public static Updater toObject(PropertySet properties) {
    return toBasket(Basket.ofObject(properties));
  }

}

File: raffia-core/src/main/java/com/codepoetics/raffia/operations/Setters.kt
package com.codepoetics.raffia.operations

import com.codepoetics.raffia.baskets.ArrayContents
import com.codepoetics.raffia.baskets.Basket
import com.codepoetics.raffia.baskets.ObjectEntry
import com.codepoetics.raffia.baskets.PropertySet
import com.codepoetics.raffia.functions.Updater

import java.math.BigDecimal

object Setters {

    @JvmStatic
    fun toBasket(value: Basket): Updater {
        return Updaters.toConstant(value)
    }

    @JvmStatic
    fun toString(value: String): Updater {
        return toBasket(Basket.ofString(value))
    }

    @JvmStatic
    fun toNumber(value: BigDecimal): Updater {
        return toBasket(Basket.ofNumber(value))
    }

    @JvmStatic
    fun toBoolean(value: Boolean): Updater {
        return toBasket(Basket.ofBoolean(value))
    }

    @JvmStatic
    fun toNull(): Updater {
        return toBasket(Basket.ofNull())
    }

    @JvmStatic
    fun toArray(contents: Collection<Basket>): Updater {
        return toBasket(Basket.ofArray(contents))
    }

    @JvmStatic
    fun toArray(contents: ArrayContents): Updater {
        return toBasket(Basket.ofArray(contents))
    }

    @JvmStatic
    fun toArray(vararg contents: Basket): Updater {
        return toBasket(Basket.ofArray(*contents))
    }

    @JvmStatic
    fun toObject(vararg properties: ObjectEntry): Updater {
        return toBasket(Basket.ofObject(*properties))
    }

    @JvmStatic
    fun toObject(properties: Map<String, Basket>): Updater {
        return toBasket(Basket.ofObject(properties))
    }

    @JvmStatic
    fun toObject(properties: PropertySet): Updater {
        return toBasket(Basket.ofObject(properties))
    }

}
